### PR TITLE
Changed most info-level log messags in gc_worker.rs to debug-level

### PR DESCRIPTION
**Pull request by kolbe**
## What have you changed? (mandatory)

I changed most info-level log messages in src/storage/gc_worker.rs to be debug-level instead of info-level.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

My TiKV Server log was getting a couple thousand info-level messages from gc_worker.rs after running the mysql-test suite against TiDB Server. It creates a huge number of tables, inserts a small amount of data into each of them, then drops them. I guess that left lots of garbage for TiKV to clean up, and it was all too happy to verbosely describe its work with hundreds of "unsafe destroy range" and "GC found/deleted plenty versions for a key" messages, among others.

I don't think this is info that a typical user or administrator of TiKV Server needs to consume, so I think the "info" level for these messages is incorrect.

I kept "start auto gc" and "finished auto gc" at info-level, since perhaps it is useful for an administrator to be able to see that gc is functioning. If not, those could also be moved to debug-level.

## How has this PR been tested? (mandatory)

I was able to successfully build tikv-server after making these changes, and I confirmed that the messages are no longer displayed when logging at info-level, but *are* displayed when logging at debug-level.

## Does this PR affect documentation (docs) update? (mandatory)
N/A.

## Does this PR affect tidb-ansible update? (mandatory)
No.

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @zhangjinpeng1987 @MyonKeminta

File: src/storage/gc_worker.rs [hunk 1]
```diff
@@ -746,9 +746,9 @@ impl<S: GCSafePointProvider, R: RegionInfoProvider> GCManager<S, R> {
     /// of `safe_point`. TiKV won't do any GC automatically until the first time `safe_point` was
     /// updated to a greater value than initial value.
     fn initialize(&mut self) -> GCManagerResult<()> {
-        info!("gc-manager is initializing");
+        debug!("gc-manager is initializing");
         self.safe_point = 0;
-        self.try_update_safe_point();
+        self.wait_for_next_safe_point()?;
```
Comment (MyonKeminta) [inline src/storage/gc_worker.rs]: This line is changed recently in master and I guess you've changed it back by mistake.

**Comment by MyonKeminta** [comment]
Please sign off all your commits. Thanks!

File: src/storage/gc_worker.rs (hunk 1 above)
Comment (kolbe) [reply src/storage/gc_worker.rs]: Oops, fixed.

**Comment by kolbe** [comment]
@MyonKeminta if you think some of these message should stay at the info level, that's fine with me. Maybe it would make more sense for you to make a quick run through this file to change the messages you think should be debug instead of info, since you know much more about the criticality of each operation and whether an administrator should know about them in the general course of the server running.

**Comment by MyonKeminta** [comment]
After some discussion, this PR will be closed and I'll start a new PR to do this.

**Pull request closed without merging**